content/en/blog/cdk-pipeline/index.md

```diff
@@ -0,0 +1,232 @@
+these assets using its buckets and ECR registries. By default, they have no lifecycle policies,
+so the CDK developer should ensure that the assets will not increase their AWS bill unexpectedly.
+
+[Assets building](https://github.com/findy-network/findy-agent-infra/blob/master/aws-ecs/lib/frontend.ts#L47)
+utilizes `aws-s3-deployment` module for the frontend application:
+
+```ts
+    // Source bundle
+    const srcBundle = s3deploy.Source.asset('../../findy-wallet-pwa', {
+      bundling: {
+        command: [
+          'sh', '-c',
+          'npm ci && npm run build && ' +
+          'apk add bash && ' +
+          `./create-set-env.sh "./tools/env-docker/set-env.sh" "${bucketName}" "${process.env.API_SUB_DOMAIN_NAME}.${process.env.DOMAIN_NAME}" "${GRPCPortNumber}" && ` +
+          'cp -R ./build/. /asset-output/'
+        ],
+        image: DockerImage.fromRegistry('public.ecr.aws/docker/library/node:18.12-alpine3.17'),
+        environment: {
+          REACT_APP_GQL_HOST: bucketName,
+          REACT_APP_AUTH_HOST: bucketName,
+          REACT_APP_HTTP_SCHEME: 'https',
+          REACT_APP_WS_SCHEME: 'wss',
+        },
+      },
+    });
+
+    new s3deploy.BucketDeployment(this, `${id}-deployment`, {
+      sources: [srcBundle],
+      destinationBucket: bucket,
+      logRetention: RetentionDays.ONE_MONTH
+    });
+```
+
+### 5/5 Deploy
+
+Finally, the Deploy stage creates and updates the application infrastructure resources.
+There is also a chance to add post steps to this stage, which can run the post-deployment
+testing and other needed scripts.
+
+{{< imgproc deploy Fit "925x925" >}}
+{{< /imgproc >}}
+
+For the agency, we are using the post-deployment steps for three purposes.
+Firstly we have a custom script for updating the ECS service. This script is in place
+to tweak some service parameters missing from CDK constructs. Secondly, we do
+the configuration of the agency administrator account. And last, we are running
+an e2e test round to ensure the deployment was successful.
+
+## Conclusions
+
+The CDK pipeline is initially a bit complex to get your head around. For simple applications,
+the use is easy, and there isn't even a need to deeply understand how it works.
+However, when the deployment has multiple moving parts, it is beneficial to understand
+the different stages.
+
+There are still some details that I would like to see improvement in.
+The documentation and examples need additions, especially on how to use the assets correctly.
+There have been improvements already, but complete example applications would make
+the learning curve for the CDK pipelines more gentle. They state that CDK pipelines are
+"opinionated," but users should know better what that opinion is.
+
+However, the CDK pipeline model pleases me in many ways. I especially value
+the approach that has minimized the steps needed to run in the developer's local environment.
```

Comment:
An enlightened or skeptical reader might wonder: minimized compared to what? And they need to be reminded of that...